### PR TITLE
Use router state to communicate `mode` value to cluster wizard

### PR DESCRIPTION
**Pull request by Waseem826**
**What this PR does / why we need it**:
Use router state to communicate `mode` value to cluster wizard. Previously we were using query params for `mode` which is not ideal since that information is not important for end user.

**What type of PR is this?**

/kind bug

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
NONE
```

**Documentation**:

```documentation
NONE
```

### CODE REVIEW

**Review by ahmedwaleedmalik — APPROVED** [review]
/approve

**Comment by ahmedwaleedmalik** [comment]
@Waseem826 I think we should merge this before the release. So maybe it should have `code-freeze-approved` label :D

**Comment by Waseem826** [comment]
@ahmedwaleedmalik I'll add it now. I was waiting for the approval 😉

**Pull request merged**
